### DISCUSSION TITLE
Micro-mechanical-informed creep model

### DISCUSSION THREAD

**Discussion by RossMech** [idaholab/moose#24627] [Q&A Modules: Solid mechanics]
Hello everyone,

I would like to build the analytically homogenized creep model for lamellar structures.

The idea is to calculate stress using the anisotropic interaction law and the plastic strains inside the phases using radial return law.

If someone has an idea for starting point, I would be very greatful.

P.S. I have an experience in UMAT writing for Abaqus, maybe it would be a good idea to write UMAT instead of the modifying the MOOSE subroutines?

**Comment by GiudGiud** [comment]
We do support UMAT in MOOSE too
@dschwen will guide you further

**Reply by RossMech** [reply]
Thank you for the answer! 
I find Materials in the TensorMechanics a little bit to convoluted and not so clear for understanding, maybe it would really an optimal solution to implement the UMAT instead of modification of existing MOOSE creep models

**Comment by ngrilli** [comment]
@RossMech 
I have recently implemented a physics-based creep model by deriving from RadialReturnCreepStressUpdateBaseTempl,
see an example here:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/SA508CreepStressUpdate.C

which is a simplification of the model in Lu et al. (2019) for ferritic steel.

Basically to create your own creep model you need to define a specific "creep_rate"
in the function computeResidualInternal
and its derivative with respect to Mises stress "creep_rate_derivative"
in the function computeDerivative

**Comment by RossMech** [comment]
Hello everyone, 

thank you for the answers. I would like to share my experience here with you. 

As I said, the creep model was micro-mechanically informed. Therefore, the elastic stiffness should also be recalculated here. Therefore, I used computestress class and rewritten some of the methods there to implement my model. 

Due to the incrporation of the tensor algebra features in RankTwoTensor and RankFourTensor methods, the implementation was much easier in comparison to UMAT programming (I personally can not bear the Voigt representation of stress, strain and stiffness tensors). 

I would share more details as we get the research project and the publication done. 

I would like to thank the community and the developers for such good tensor mechanics module infrastructure!

**Reply by GiudGiud** [reply]
Hello

Thank you for the feedback. I'm sure @dschwen and @amjokisaari (and all the others involved in the development of these methods) will be happy to hear this. 

Guillaume
